src/index.ts

```diff
@@ -1,50 +1,35 @@
 import { ComponentSettings, Manager, MCEvent } from '@managed-components/types'
 
-interface RedditPayload {
+// For future reference, see payload being sent here: https://reddit-test-2.pages.dev/
+type RedditPayload = {
   id: string
-  event: object
-  ts: number
+  event: string
+  ts: string
   click_id?: string
   uuid?: string
-}
-
-export const redditHashKey = 'm4sxh0fc5dnb' // Hardcoded Reddit key
-
-const redditHash = async (data: RedditPayload) => {
-  let message = data.id + data.event + data.ts
-
-  if (data.click_id) message += data.click_id
-  if (data.uuid) message += data.uuid
-
-  const getUtf8Bytes = (str: string) =>
-    new Uint8Array(
-      [...unescape(encodeURIComponent(str))].map(c => c.charCodeAt(0))
-    )
-
-  const keyBytes = getUtf8Bytes(redditHashKey)
-  const messageBytes = getUtf8Bytes(message)
-
-  const cryptoKey = await crypto.subtle.importKey(
-    'raw',
-    keyBytes,
-    { name: 'HMAC', hash: 'SHA-256' },
-    true,
-    ['sign']
-  )
-  const sig = await crypto.subtle.sign('HMAC', cryptoKey, messageBytes)
-
-  // to lowercase hexits
-  ;[...new Uint8Array(sig)].map(b => b.toString(16).padStart(2, '0')).join('')
-
-  // to base64
-  return btoa(String.fromCharCode(...new Uint8Array(sig)))
+  integration?: string
+  opt_out?: string
+  v?: string
+  sh?: string
+  sw?: string
+  'm.itemCount'?: string
```

Comment:
Should this be a string containing a "." (dot) or should this rather be the key "m" containing an object of key/values?

Simplified example:
```
type RedditPayload = {
  ...
  m: {
    itemCount?: string
    ...
  }
}
```

Reply:
I can confirm I tried the object approach, but `URLSearchParams` types the object to string like this:
<img width="342" alt="image" src="https://user-images.githubusercontent.com/47219640/225625519-8296ea91-cc18-435b-be47-1bc36dc910f3.png">